### ISSUE TITLE
Allow sasl_mechanism to be specified for kafka event brokers

### ISSUE THREAD

**Issue by jabel415** [RasaHQ/rasa#8745]
**Description of Problem**:
`sasl_mechanism` is hardcoded to `PLAIN` for Kafka Event Brokers that use SASL_SSL security protocol

I am trying to connect to an MSK Cluster for my event broker, but it requires the SASL Mechanism SCRAM-SHA-512, as per https://docs.aws.amazon.com/msk/latest/developerguide/msk-password.html (Step 6). Currently I receive the following in my MSK logs:

`[2021-05-25 13:04:42,851] INFO [SocketServer brokerId=2] Failed authentication with INTERNAL_IP/INTERNAL_IP (Unsupported SASL mechanism PLAIN) (org.apache.kafka.common.network.Selector)`

**Overview of the Solution**:
Add sasl_mechanism as a configurable parameter for the KafkaEventBroker that can be leveraged in the event_broker definition within endpoints.yml

**Examples** (if relevant):
Add `sasl_mechanism: Optional[Text] = None,` after https://github.com/RasaHQ/rasa/blob/7625fcaec9a5d1f47dbcc093a405e9369e7261fc/rasa/core/brokers/kafka.py#L22 .
Change https://github.com/RasaHQ/rasa/blob/7625fcaec9a5d1f47dbcc093a405e9369e7261fc/rasa/core/brokers/kafka.py#L172 to `sasl_mechanism=self.sasl_mechanism`

**Definition of Done**:
- [ ] Tests are added
- [ ] Update docs here https://rasa.com/docs/rasa/event-brokers/#adding-a-kafka-event-broker-using-the-endpoint-configuration to show example value added in "If using the SASL_SSL protocol"
- [ ] Feature mentioned in the changelog

**Comment by sara-tagger** [comment]
Thanks for submitting this feature request 🚀 @kedz will get back to you about it soon! ✨

**Comment by kedz** [comment]
@jabel415 thank you for bringing this to our attention! Looks like there is [another PR](https://github.com/RasaHQ/rasa/pull/8724) that is almost ready to be merged which will solve this issue. Please correct me though if I'm mistaken.

**Comment by jabel415** [comment]
@kedz Thanks for the reply. Yes that is correct, I’ve been tracking the PR with @mvielkind and was planning to close this upon merge of that PR. Looks like it’s waiting on review from @wochinge

**Comment by b-quachtran** [comment]
@jabel415 Closing this issue as the PR has just been merged into main
